src/opendlv-video-h264-encoder.cpp

```diff
@@ -83,31 +137,63 @@ int32_t main(int32_t argc, char **argv) {
-                parameters.iEntropyCodingModeFlag = 0; // 0 = CAVLC, 1 = CABAC (not supported in BaseLine profile).
-                parameters.iComplexityMode = ECOMPLEXITY_MODE::LOW_COMPLEXITY;
-                parameters.bEnableAdaptiveQuant = 1;
-                parameters.bEnableBackgroundDetection = 1;
-                parameters.bEnableDenoise = 1;
-                parameters.bEnableFrameSkip = 0;
-                parameters.bEnableLongTermReference = 0;
-                parameters.bEnableSceneChangeDetect = 1;
-                parameters.bPrefixNalAddingCtrl = 0; // do nod add NAL prefixes.
-                parameters.eSpsPpsIdStrategy = EParameterSetStrategy::CONSTANT_ID;
+                parameters.iMultipleThreadIdc = I_MULTIPLE_THREADS; // 1 = disable multi threads.
 
                 parameters.sSpatialLayers[0].iVideoWidth = parameters.iPicWidth;
                 parameters.sSpatialLayers[0].iVideoHeight = parameters.iPicHeight;
                 parameters.sSpatialLayers[0].fFrameRate = parameters.fMaxFrameRate;
                 parameters.sSpatialLayers[0].iSpatialBitrate = parameters.iTargetBitrate;
-                parameters.sSpatialLayers[0].iMaxSpatialBitrate = parameters.iMaxBitrate;
+                parameters.sSpatialLayers[0].iMaxSpatialBitrate = I_BITRATE_MAX;
                 parameters.sSpatialLayers[0].sSliceArgument.uiSliceMode = SliceModeEnum::SM_SIZELIMITED_SLICE;
                 parameters.sSpatialLayers[0].sSliceArgument.uiSliceNum = 1;
+
+                /*
+                 * Thesis parameters
+                 * https://github.com/cisco/openh264/wiki/TypesAndStructures
+                 * https://github.com/cisco/openh264/blob/master/codec/encoder/core/inc/param_svc.h#L132
+                 */
+                if (I_NUM_REF_FRAME==0){parameters.iNumRefFrame = AUTO_REF_PIC_COUNT;}
+                else {parameters.iNumRefFrame = I_NUM_REF_FRAME;}
+                parameters.bPrefixNalAddingCtrl = B_PREFIX_NAL;
+                parameters.bEnableSSEI = B_SSEI;
+                parameters.iPaddingFlag = I_PADDING;
+                parameters.iEntropyCodingModeFlag = I_ENTROPY_CODING;
+                parameters.bEnableFrameSkip = B_FRAME_SKIP;
+                parameters.iMaxBitrate = I_BITRATE_MAX;
+                parameters.iMaxQp = I_MAX_QP;
+                parameters.iMinQp = I_MIN_QP;
+                parameters.bEnableLongTermReference = B_LONG_TERM_REFERENCE;
+                parameters.iLoopFilterDisableIdc = I_LOOP_FILTER;
+                parameters.bEnableDenoise = B_DENOISE;
+                parameters.bEnableBackgroundDetection = B_BACKGROUND_DETECTION;
+                parameters.bEnableAdaptiveQuant = B_ADAPTIVE_QUANT;
+                parameters.bEnableFrameCroppingFlag = B_FRAME_CROPPING;
+                parameters.bEnableSceneChangeDetect = B_SCENE_CHANGE_DETECT;
+
+                switch (RC_MODE) {
+                    case 0: {parameters.iRCMode = RC_MODES::RC_QUALITY_MODE; break;}
```

Comment:
Please add a blank after `{` and before `}`

Reply:
Fixed in 7a978cd